ibis/backends/tests/test_register.py

```diff
@@ -594,3 +614,73 @@ def test_read_csv(con, data_dir, in_table_name, num_diamonds):
         }
     )
     assert table.count().execute() == num_diamonds
+
+
+@pytest.mark.parametrize(
+    ("skip_rows", "new_column_names", "delimiter", "include_columns"),
+    [
+        param(True, True, False, False, id="skip_rows_with_column_names"),
+        param(False, False, False, True, id="include_columns"),
+        param(False, False, True, False, id="delimiter"),
+    ],
+)
+@pytest.mark.notyet(["flink"])
+@pytest.mark.notimpl(["druid"])
+@pytest.mark.never(
+    [
+        "duckdb",
+        "polars",
+        "bigquery",
+        "clickhouse",
+        "datafusion",
+        "snowflake",
+        "pyspark",
+    ],
+    reason="backend implements its own read_csv",
+)
+@pytest.mark.notimpl(["mssql"], raises=PyODBCProgrammingError)
+@pytest.mark.notimpl(["mysql"], raises=MySQLOperationalError)
+def test_read_csv_pyarrow_options(
+    con, tmp_path, ft_data, skip_rows, new_column_names, delimiter, include_columns
+):
+    pc = pytest.importorskip("pyarrow.csv")
+
+    if con.name in (
+        "duckdb",
+        "polars",
+        "bigquery",
+        "clickhouse",
+        "datafusion",
+        "snowflake",
+        "pyspark",
+    ):
+        pytest.skip(f"{con.name} implements its own `read_parquet`")
```

Comment:
these backends have their own implementation, some of these options still could pass this test, so I skip these backends.